core/src/main/java/com/sequenceiq/cloudbreak/service/image/userdata/UserDataService.java

```diff
@@ -89,24 +101,54 @@ public void updateProxyConfig(Long stackId) {
         String gatewayUserData = userDataByInstanceGroup.get(InstanceGroupType.GATEWAY);
         String result = proxyConfigUserDataReplacer.replaceProxyConfigInUserDataByEnvCrn(gatewayUserData, stack.getEnvironmentCrn());
         userDataByInstanceGroup.put(InstanceGroupType.GATEWAY, result);
-        updateUserData(stack, userDataByInstanceGroup);
+        updateUserData(stack.getId(), userDataByInstanceGroup);
     }
 
-    private Map<InstanceGroupType, String> getUserData(Long stackId) {
+    public Map<InstanceGroupType, String> getUserData(Long stackId) {
         try {
             Image image = imageService.getImage(stackId);
-            return new HashMap<>(image.getUserdata());
+            if (image.getUserdata() == null || image.getUserdata().isEmpty()) {
+                Map<InstanceGroupType, String> map = new HashMap<>();
+                Optional<Userdata> userdataOptional = userdataRepository.findByStackId(stackId);
+                if (userdataOptional.isPresent()) {
```

Comment:
If this optional is empty then it signals an error, that is, no userdata found. Still, the method returns with an empty map.

Reply:
on old cluster the image object will contains userdata and on new cluster it is always in the userdata table.